pode/geometry_utils.py

```diff
@@ -35,14 +33,19 @@
 
 
 def segments(ring: LinearRing) -> Iterator[LineString]:
-    """Yields consecutive lines from the given ring"""
+    """
+    Yields consecutive lines from the given ring
+    Doesn't work for degenerate geometries.
+    """
     pairs = pairwise(ring.coords)
     yield from map(LineString, pairs)
 
 
 def right_left_parts(polygon: Polygon,
                      line: LineString) -> Tuple[Polygon, Polygon]:
     """Splits polygon by a line and returns two parts: right and left"""
+    if len(line.coords) > 2:
+        raise ValueError("Only lines consisting of 2 points are supported")

```

Comment:
can line consist of 0 points? we probably can change condition to
```python
len(line.coords) != 2
```

Reply:
In Shapely yes, a line can consist of zero points. I'll update this condition, thanks!